pandas/tests/series/methods/test_astype.py

```diff
@@ -446,7 +446,7 @@ def test_astype_string_to_extension_dtype_roundtrip(
         self, data, dtype, request, nullable_string_dtype
     ):
         if dtype == "boolean" or (
-            dtype in ("period[M]", "datetime64[ns]", "timedelta64[ns]") and NaT in data
```

Comment:
Partially addresses #40566 by fixing handling of pd.NA for conversion to period types.